src/main/java/com/internet/shop/security/AuthenticationImpl.java

```diff
@@ -8,17 +8,16 @@
 
 @Service
 public class AuthenticationImpl implements AuthenticationService {
-    private static final String ERROR_MESSAGE = "Incorrect login or password.";
     @Inject
     private UserService userService;
 
     @Override
     public User login(String login, String password) throws AuthenticationException {
         User userFromDb = userService.findByLogin(login).orElseThrow(() ->
-                new AuthenticationException(ERROR_MESSAGE));
+                new AuthenticationException("Incorrect login or password."));
         if (userFromDb.getPassword().equals(password)) {
```

Comment:
let's check all conditions inside `if` so that we could throw our Exception only once